x-pack/test/functional/apps/discover/feature_controls/discover_spaces.ts

```diff
@@ -176,6 +176,7 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
         await PageObjects.common.navigateToUrl('discover', '', {
           basePath: '/s/custom_space_no_index_patterns',
           ensureCurrentUrl: false,
+          shouldUseHashForSubUrl: false,
```

Comment:
QQ: why is this change needed?

Reply:
the test was creating an incorrect URL when trying to navigate to Discover

Reply:
oh, I see, guess it's a legacy url , which is a indicator that they don't work anymore. @timroes should we still support these?

Reply:
Yes, I think we need to continue supporting those legacy URLs still for some while, before we at least show some notification in the UI that they are deprecated when users are using them. We're working on some larger changes, to make it clear how we communicate deprecated URLs when users are opening them to give them a chance to update them, but it's not ready yet. Before we have that, I don't think we can silently drop support for the legacy URLs. And given that we see it's easy to accidentally modify a test that implicitly tested that, I'd recommend we add some more explicit test for legacy URL behavior as part of that PR.

Reply:
so we need to have some redirection urls to be built in ... didn't check but this might be hashed URL like `#discover/....`

Reply:
it's not a legacy URL, it's just the way the test constructs the URL was not correct. when manually navigating to the URL, everything works fine.

Reply:
the thing is that redirect works correctly if you navigate manually. I think this is just due to the way tests handle navigation, I'm sure we've had a case like this in the past. let me see if I can try and find it.